### PR TITLE
Added Arch recipe

### PR DESCRIPTION
**Pull request by DanielBok**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by DanielBok** [comment]
@conda-forge/core Like your help in understanding why there is an ImportError for the test phase in the OSX build.

**Comment by isuruf** [comment]
@conda-forge-linter, relint

File: recipes/arch/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+{% set name = "arch" %}
```
Comment (isuruf) [inline recipes/arch/meta.yaml]: I'm not sure about using this name. Other package managers use this name for another package.

Comment (DanielBok) [reply recipes/arch/meta.yaml]: Understand your concern. I changed the package name to `arch-py` and left the `name` there to refer to the pypi file path.

```
package:
    name: arch-py
```

I take it that when other users install the package this way, there shouldn't be any conflict.

Comment (DanielBok) [reply recipes/arch/meta.yaml]: @isuruf, ping me if you still have any lingering concerns :)

**Comment by DanielBok** [comment]
@conda-forge/help-python ready for review again :)

**Comment by DanielBok** [comment]
@conda-forge/staged-recipes bump

**Comment by DanielBok** [comment]
Thank you. :)

**Pull request merged**